### ISSUE TITLE
Lbank watchOrderBook breaks on second ping-pong

### ISSUE THREAD

**Issue by markant722** [ccxt/ccxt#23562]
### Operating System

Ubuntu

### Programming Languages

JavaScript

### CCXT Version

4.3.86

### Description

Hi guys,

For lbank exchange pro version, when I watch pair BTC/USDT after 2 or 3 ping pongs web socket breaks, I was looking at their docs you did right thing with code I cant find error, what could be cause of this?

Should I write to lbank? I am giving you snippet of log what happens on second ping/pong, We receive ping and send pong and it breaks immediately:

`2024-08-27T21:33:45.444Z onMessage { ping: '21c2b207-44eb-46c2-87b4-0fdf9eb982d7', action: 'ping' }
2024-08-27T21:33:45.446Z sending { action: 'pong', pong: '21c2b207-44eb-46c2-87b4-0fdf9eb982d7' }
2024-08-27T21:33:45.448Z onClose CloseEvent {
  [Symbol(kTarget)]: <ref *1> WebSocket {
    _events: [Object: null prototype] {
      open: [Function],
      message: [Function],
      error: [Function],
      close: [Function],
      ping: [Function: bound onPing],
      pong: [Function: bound onPong],
      upgrade: [Function: bound onUpgrade]
    },
`

Can you help me how could I set streaming and custom ping, maybe we should do keepAlive on our own to ping it more frequently?

### Code

```
  

```

**Comment by sc0Vu** [comment]
Hi @markant722 It works for me. Can you check your internet condition or paste your code?

**Comment by markant722** [comment]
It happens when difference between to pings is bigger then 60 seconds. This is last example it ping-pongs:
2024-08-28T00:05:12.161Z onMessage { ping: 'b80c4d3e-1cb3-4160-abeb-e3d0037347b9', action: 'ping' }
2024-08-28T00:05:12.163Z sending { action: 'pong', pong: 'b80c4d3e-1cb3-4160-abeb-e3d0037347b9' }

Then on next ping from lbank I send pong but its closed, because I received that ping slightly above 60 sec interval:
2024-08-28T00:06:12.619Z onMessage { ping: 'b2ab3d67-8fd0-4d58-8b18-b56ec3e83a0a', action: 'ping' }
2024-08-28T00:06:12.622Z sending { action: 'pong', pong: 'b2ab3d67-8fd0-4d58-8b18-b56ec3e83a0a' }
2024-08-28T00:06:12.624Z onClose CloseEvent {
  [Symbol(kTarget)]: <ref *1> WebSocket {
    _events: [Object: null prototype] {
      open: [Function],
      message: [Function],
      error: [Function],
      close: [Function],
      ping: [Function: bound onPing],
      pong: [Function: bound onPong],
      upgrade: [Function: bound onUpgrade]
    },


I tried to add manual ping from my side in file node_modules/ccxt/dist/cjs/src/pro/lbank.js changed 
'streaming': {
                 'ping': this.ping,
                 'keepAlive': 10000,
            },

Then added 

     async ping(client) {
         if (typeof this.pingId !== 'undefined'){
    
             await client.send({'action':'ping','ping':this.pingId});
    
         }
     }

And changed method to keep pingId as :
async handlePing(client, message) {
        //
        //  { ping: 'a13a939c-5f25-4e06-9981-93cb3b890707', action: 'ping' }
        //
        const pingId = this.safeString(message, 'ping');
        this.pingId = pingId;
        await client.send({
            'action': 'pong',
            'pong': pingId,
        });
    }


But Lbank returns to me:

2024-08-28T00:12:45.967Z sending { action: 'ping', ping: 'e31ee860-c628-4a1a-825d-92345590588d' }
2024-08-28T00:12:45.968Z sending Promise { <pending> }
2024-08-28T00:12:46.172Z onMessage { action: 'pong', pong: 'e31ee860-c628-4a1a-825d-92345590588d' }
2024-08-28T00:12:46.173Z onMessage {
  SERVER: 'V2',
  message: "Missing parameter ['action']",
  status: 'error',
  TS: '2024-08-28T08:12:46.061'
}

I have included action, dont know why it says action missing.....

**Comment by sc0Vu** [comment]
@markant722 We've implemented ping in lbank, it should work like they mentioned in documentation: https://www.lbank.com/en-US/docs/index.html#request-amp-subscription-instruction.

After test, it's weird the server close the connection even we sent the pong in 1 minute. Probably need to ask them.

```
2024-08-28T01:55:44.742Z onMessage { ping: 'd6102cb9-8271-45eb-8b30-f296408c02dd', action: 'ping' }
2024-08-28T01:55:44.749Z sending { action: 'pong', pong: 'd6102cb9-8271-45eb-8b30-f296408c02dd' }
```

**Comment by rajnikyadav** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by rajnikyadav** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by ViniciusSCG** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by markant722** [comment]
Yes, I got to same conclusion, even I send regular pings on regular intervals server closes connection with:
`2024-08-28T09:44:58.835Z onError Connection to wss://www.lbkex.net/ws/V2/ timed out due to a ping-pong keepalive missing on time
`

**Comment by markant722** [comment]
I just want to mention that on version 4.4.40 this bug is solved
